app/src/test/java/com/kickstarter/viewmodels/ProjectViewModelTest.kt

```diff
@@ -510,40 +510,54 @@ class ProjectViewModelTest : KSRobolectricTestCase() {
     }
 
     @Test
-    fun testManagePledgeMenuIsVisible_whenProjectBacked() {
+    fun testManagePledgeMenu_whenProjectBackedAndLive() {
         setUpEnvironment(environmentWithNativeCheckoutEnabled())
 
         // Start the view model with a backed project
         this.vm.intent(Intent().putExtra(IntentKey.PROJECT, ProjectFactory.backedProject()))
 
-        this.managePledgeMenuIsVisible.assertValue(true)
+        this.managePledgeMenu.assertValue(R.menu.manage_pledge_live)
     }
 
     @Test
-    fun testManagePledgeMenuIsVisible_whenProjectNotBacked() {
+    fun testManagePledgeMenu_whenProjectBackedAndNotLive() {
+        setUpEnvironment(environmentWithNativeCheckoutEnabled())
+
+        // Start the view model with a backed project
+        val successfulBackedProject = ProjectFactory.backedProject()
+                .toBuilder()
+                .state(Project.STATE_SUCCESSFUL)
+                .build()
+        this.vm.intent(Intent().putExtra(IntentKey.PROJECT, successfulBackedProject))
+
+        this.managePledgeMenu.assertValue(R.menu.manage_pledge_ended)
+    }
+
+    @Test
+    fun testManagePledgeMenu_whenProjectNotBacked() {
         setUpEnvironment(environmentWithNativeCheckoutEnabled())
 
         // Start the view model with a backed project
         this.vm.intent(Intent().putExtra(IntentKey.PROJECT, ProjectFactory.project()))
 
-        this.managePledgeMenuIsVisible.assertValue(false)
+        this.managePledgeMenu.assertValue(null)
     }
 
     @Test
-    fun testManagePledgeMenuIsVisible_whenManaging() {
+    fun testManagePledgeMenu_whenManaging() {
         setUpEnvironment(environmentWithNativeCheckoutEnabled())
 
         // Start the view model with a backed project
         this.vm.intent(Intent().putExtra(IntentKey.PROJECT, ProjectFactory.backedProject()))
 
-        this.managePledgeMenuIsVisible.assertValue(true)
+        this.managePledgeMenu.assertValue(R.menu.manage_pledge_live)
 
         this.vm.inputs.cancelPledgeClicked()
         this.vm.inputs.fragmentStackCount(1)
-        this.managePledgeMenuIsVisible.assertValues(true, false)
+        this.managePledgeMenu.assertValues(R.menu.manage_pledge_live, null)
 
         this.vm.inputs.fragmentStackCount(0)
-        this.managePledgeMenuIsVisible.assertValues(true, false, true)
+        this.managePledgeMenu.assertValues(R.menu.manage_pledge_live, null, R.menu.manage_pledge_live)
```

Comment:
These tests overall look good! Just for my understanding, is this saying that these function was called with all three of these inputs at once, or in this order? Seeing the type is:

```
fun managePledgeMenu(): Observable<Int?>
```

Out of curiosity what testing library is used for the assertions?

Reply:
In this order! That's the stream of emissions.

We use http://robolectric.org/ to test.